test/sqllogictest/order_by.slt

```diff
@@ -1174,3 +1174,409 @@ Used Indexes:
+# Tests for prepared statement parameters in OFFSET, and for non-trivial expressions in OFFSET.
+# (Non-trivial expressions in OFFSET clauses have to be simplifiable to a literal, possibly after parameter binding.)
+#
+# (LIMIT clauses with prepared statement parameters have tests in `test_bind_params` in `pgwire.rs`.
+# LIMIT clauses with non-trivial expressions, referring to the outer context, have tests in `limit_expr.slt`.)
+########################################################################################################################
+
+statement ok
+PREPARE p1 AS
+SELECT *
+FROM foo
+ORDER BY a, b
+OFFSET $1;
+
+query IT
+EXECUTE p1(0::bigint);
+----
+0  zero
+1  one
+2  two
+
+query IT
+EXECUTE p1(1::bigint);
+----
+1  one
+2  two
+
+query IT
+EXECUTE p1((1+1)::bigint);
+----
+2  two
+
+query error db error: ERROR: Invalid OFFSET clause: Expected an expression that evaluates to a non\-null value, got null
+EXECUTE p1(null::bigint);
+
+query error db error: ERROR: Invalid OFFSET clause: Expected an expression that evaluates to a non\-null value, got null
+PREPARE p_error AS
+SELECT *
+FROM foo
+ORDER BY a, b
+OFFSET null;
+
+# Prepared statement parameter in OFFSET inside a subquery
+statement ok
+PREPARE p2 AS
+SELECT
+  (SELECT sum(a) FROM (
+    SELECT a
+    FROM foo
+    ORDER BY a, b
+    OFFSET $1
+  ))
+FROM foo AS outer_foo
+OFFSET $2;
+
+query I
+EXECUTE p2(0::bigint, 0::bigint);
+----
+3
+3
+3
+
+query I
+EXECUTE p2(2::bigint, 0::bigint);
+----
+2
+2
+2
+
+query I
+EXECUTE p2(0::bigint, 1::bigint);
+----
+3
+3
+
+query I
+EXECUTE p2(2::bigint, 1::bigint);
+----
+2
+2
+
+statement ok
+PREPARE p3 AS
+SELECT *
+FROM foo
+ORDER BY a, b
+OFFSET $1 - 2;
+
+query error db error: ERROR: Invalid OFFSET clause: must not be negative, got \-2
+EXECUTE p3(0);
+
+query II
+EXECUTE p3(2);
+----
+0  0
+1  0
+2  0
+
+query II
+EXECUTE p3(3);
+----
+1  0
+2  0
+
+query error db error: ERROR: mismatched parameter type: expected bigint, got integer
+EXECUTE p1(1);
+
+query error db error: ERROR: mismatched parameter type: expected bigint, got text
+EXECUTE p1('aaa');
+
+query error db error: ERROR: Invalid OFFSET clause: must not be negative, got \-7
+PREPARE p_error AS
+SELECT *
+FROM foo
+ORDER BY a, b
+OFFSET -7;
+
+query error db error: ERROR: Invalid OFFSET clause: must not be negative, got \-2
+PREPARE p_error AS
+SELECT *
+FROM foo
+ORDER BY a, b
+OFFSET 5-7;
+
+query error db error: ERROR: Invalid OFFSET clause: invalid input syntax for type bigint: invalid digit found in string: "aaa"
+PREPARE p_error AS
+SELECT *
+FROM foo
+ORDER BY a, b
+OFFSET 'aaa';
+
+query error db error: ERROR: column "a" does not exist
+PREPARE p_error AS
+SELECT *
+FROM foo
+ORDER BY a, b
+OFFSET a;
+
+# TODO: This error msg is incomplete: it's missing the `integer_to_bigint\(\#\^0\{a\}\)` from the end, which is present
+# in the actual error msg. If we include that, then sqllogictest panics due to a bug in sqllogictest and/or in
+# `regex::escape`.
```

Comment:
I will open an issue for this, and fix it separately. (Might be as simple as adding the `x` flag when working with the resulting regex, and then `\#` will be considered a valid escape sequence.)

Reply:
(Fixed in https://github.com/MaterializeInc/materialize/pull/32447)